src/main/java/com/google/api/codegen/metacode/InitCodeNode.java

```diff
@@ -0,0 +1,406 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.metacode;
+
+import com.google.api.codegen.util.Name;
+import com.google.api.codegen.util.SymbolTable;
+import com.google.api.codegen.util.testing.TestValueGenerator;
+import com.google.api.tools.framework.model.Field;
+import com.google.api.tools.framework.model.TypeRef;
+import com.google.protobuf.DescriptorProtos.FieldDescriptorProto.Type;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/*
+ * Represents a node in an tree of objects to be initialized.
+ */
+public class InitCodeNode {
+
+  private String key;
+  private InitCodeLineType lineType;
+  private InitValueConfig initValueConfig;
+  private Map<String, InitCodeNode> children;
+  private TypeRef typeRef;
+  private Name identifier;
+  private InitCodeLine initCodeLine;
```

Comment:
I wanted to... but the problem is that we need to modify theses values after construction in the update() function, and AutoValue doesn't allow that.